pkg/kine/drivers/generic/generic.go

```diff
@@ -337,8 +352,17 @@ func (d *Generic) query(ctx context.Context, txName, sql string, args ...interfa
 
 func (d *Generic) queryPrepared(ctx context.Context, txName, sql string, prepared *sql.Stmt, args ...interface{}) (result *sql.Rows, err error) {
 	logrus.Tracef("QUERY %v : %s", args, Stripped(sql))
+
+	done, err := d.ac.Admit(ctx)
+	if err != nil {
+		err = fmt.Errorf("denied: %w", err)
+		return
```

Comment:
also in other places 
```suggestion
		return nil, fmt.Errorf("denied: %w", err)
```